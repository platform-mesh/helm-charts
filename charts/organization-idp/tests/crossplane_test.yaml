suite: crossplane
release:
  name: keycloak
  namespace: openmfp-system
templates:
  - crossplane/*
set:
  crossplane:
    enabled: true
tests:
  - it: matches the snapshot
    asserts:
      - matchSnapshot: {}
  - it: matches the snapshot with custom config
    set:
      crossplane:
        identityProviders:
          custom-provider:
            authorizationUrl: https://custom-provider.com/oauth/authorize
            tokenUrl: https://custom-provider.com/oauth/token
            issuer: https://custom-provider.com
            clientId: custom-client-id
            clientSecretRef:
              name: custom-client-secret
              key: secret-key
            extraConfig:
              CustomConfig: value
    asserts:
      - matchSnapshot: {}