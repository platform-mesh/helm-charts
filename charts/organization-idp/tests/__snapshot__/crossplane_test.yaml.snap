matches the snapshot:
  1: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: Client
    metadata:
      name: clientname
    spec:
      forProvider:
        accessType: CONFIDENTIAL
        clientId: clientname
        enabled: true
        name: ClientName
        realmIdRef:
          name: default
        standardFlowEnabled: true
        validRedirectUris:
          - http://localhost:8000/callback*
          - http://localhost:4300/callback*
      providerConfigRef:
        name: keycloak-provider-config
      writeConnectionSecretToRef:
        name: portal-client-secret-organization-idp
        namespace: openmfp-system
  2: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientDefaultScopes
    metadata:
      name: clientname-default-scopes
    spec:
      forProvider:
        clientIdRef:
          name: clientname
        defaultScopes:
          - profile
          - email
          - basic
          - acr
          - groups
          - trusted-openmfp-audiences
        realmIdRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  3: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientScope
    metadata:
      name: trusted-clientname-audiences
    spec:
      forProvider:
        includeInTokenScope: true
        name: trusted-clientname-audiences
        realmIdRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  4: |
    apiVersion: realm.keycloak.crossplane.io/v1alpha1
    kind: Realm
    metadata:
      name: default
    spec:
      forProvider:
        accessTokenLifespan: 8h
        attributes:
          organizationsEnabled: "true"
        displayName: default
        displayNameHtml: <b>default</b>
        enabled: true
        loginWithEmailAllowed: true
        realm: default
        registrationAllowed: true
        registrationEmailAsUsername: true
        ssoSessionIdleTimeout: 8h
      providerConfigRef:
        name: keycloak-provider-config
matches the snapshot with custom config:
  1: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: Client
    metadata:
      name: clientname
    spec:
      forProvider:
        accessType: CONFIDENTIAL
        clientId: clientname
        enabled: true
        name: ClientName
        realmIdRef:
          name: default
        standardFlowEnabled: true
        validRedirectUris:
          - http://localhost:8000/callback*
          - http://localhost:4300/callback*
      providerConfigRef:
        name: keycloak-provider-config
      writeConnectionSecretToRef:
        name: portal-client-secret-organization-idp
        namespace: openmfp-system
  2: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientDefaultScopes
    metadata:
      name: clientname-default-scopes
    spec:
      forProvider:
        clientIdRef:
          name: clientname
        defaultScopes:
          - profile
          - email
          - basic
          - acr
          - groups
          - trusted-openmfp-audiences
        realmIdRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  3: |
    apiVersion: oidc.keycloak.crossplane.io/v1alpha1
    kind: IdentityProvider
    metadata:
      name: custom-provider
    spec:
      forProvider:
        alias: custom-provider
        authorizationUrl: https://custom-provider.com/oauth/authorize
        clientId: custom-client-id
        clientSecretSecretRef:
          key: secret-key
          name: custom-client-secret
        defaultScopes: openid email profile
        extraConfig:
          CustomConfig: value
        hideOnLoginPage: true
        issuer: https://custom-provider.com
        realmRef:
          name: default
        tokenUrl: https://custom-provider.com/oauth/token
        trustEmail: true
      providerConfigRef:
        name: keycloak-provider-config
  4: |
    apiVersion: identityprovider.keycloak.crossplane.io/v1alpha1
    kind: IdentityProviderMapper
    metadata:
      name: lastname
    spec:
      forProvider:
        extraConfig:
          Claim: last_name
          UserAttribute: lastName
          syncMode: INHERIT
        identityProviderAlias: custom-provider
        identityProviderMapper: oidc-user-attribute-idp-mapper
        name: last_name
        realmRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  5: |
    apiVersion: identityprovider.keycloak.crossplane.io/v1alpha1
    kind: IdentityProviderMapper
    metadata:
      name: firstname
    spec:
      forProvider:
        extraConfig:
          Claim: first_name
          UserAttribute: firstName
          syncMode: INHERIT
        identityProviderAlias: custom-provider
        identityProviderMapper: oidc-user-attribute-idp-mapper
        name: first_name
        realmRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  6: |
    apiVersion: openidclient.keycloak.crossplane.io/v1alpha1
    kind: ClientScope
    metadata:
      name: trusted-clientname-audiences
    spec:
      forProvider:
        includeInTokenScope: true
        name: trusted-clientname-audiences
        realmIdRef:
          name: default
      providerConfigRef:
        name: keycloak-provider-config
  7: |
    apiVersion: realm.keycloak.crossplane.io/v1alpha1
    kind: Realm
    metadata:
      name: default
    spec:
      forProvider:
        accessTokenLifespan: 8h
        attributes:
          organizationsEnabled: "true"
        displayName: default
        displayNameHtml: <b>default</b>
        enabled: true
        loginWithEmailAllowed: true
        realm: default
        registrationAllowed: true
        registrationEmailAsUsername: true
        ssoSessionIdleTimeout: 8h
      providerConfigRef:
        name: keycloak-provider-config
